server/src/main/scala/io/delta/standalone/internal/model.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.standalone.internal
+
+import io.delta.standalone.internal.actions.{
+  Metadata => DeltaMetadata,
+  SingleAction => DeltaSingleAction
+}
+
+/**
+ * Actions defined to use in the response for delta format sharing.
+ */
+
+sealed trait DeltaResponseAction {
+  /** Turn this object to the [[DeltaFormatSingleAction]] wrap object. */
+  def wrap: DeltaResponseSingleAction
+}
+
+/**
+ * DeltaResponseProtocol which is part of the delta Protocol.
+ */
+case class DeltaResponseProtocol(minReaderVersion: Int) extends DeltaResponseAction {
+  override def wrap: DeltaResponseSingleAction = DeltaResponseSingleAction(protocol = this)
+}
+
+/**
+ * DeltaResponseFileAction used in delta sharing protocol. It wraps a delta action,
+ *   and adds 4 delta sharing related fields: id/version/timestamp/expirationTimestamp.
+ *       - id: used to uniquely identify a file, and in idToUrl mapping for executor to get
+ *             presigned url.
+ *       - version/timestamp: the version and timestamp of the commit, used to generate faked delta
+ *                            log file on the client side.
+ *       - expirationTimestamp: indicate when the presigned url is going to expire and need a
+ *                              refresh.
+ *   Suggest to redact the tags field before returning if there are sensitive info.
```

Comment:
This header comment is cleaner than the DBR version. Consider copying it there.

Reply:
done.